codec.go

```diff
@@ -145,6 +145,18 @@ func NewCodec() *Codec {
 	return cdc
 }
 
+func (cdc *Codec) TypeInfos() map[reflect.Type]*TypeInfo {
```

Comment:
I don't have too much context here, but simply returning maps on a reference type is somewhat dangerous as the caller can unexpectedly modify the reference (i.e. the codec). Instead, return a copy of the map.

Reply:
Also, let's add simple godocs.